src/contracts/LidoARM.sol

```diff
@@ -0,0 +1,124 @@
+    /// @param _name The name of the liquidity provider (LP) token.
+    /// @param _symbol The symbol of the liquidity provider (LP) token.
+    /// @param _operator The address of the account that can request and claim Lido withdrawals.
+    /// @param _fee The performance fee that is collected by the feeCollector measured in basis points (1/100th of a percent).
+    /// 10,000 = 100% performance fee
+    /// 1,500 = 15% performance fee
+    /// @param _feeCollector The account that can collect the performance fee
+    /// @param _capManager The address of the CapManager contract
+    function initialize(
+        string calldata _name,
+        string calldata _symbol,
+        address _operator,
+        uint256 _fee,
+        address _feeCollector,
+        address _capManager
+    ) external initializer {
+        _initARM(_operator, _name, _symbol, _fee, _feeCollector, _capManager);
+
+        // Approve the Lido withdrawal queue contract. Used for redemption requests.
+        steth.approve(address(lidoWithdrawalQueue), type(uint256).max);
+    }
+
+    /**
+     * @notice Request a stETH for ETH withdrawal.
+     * Reference: https://docs.lido.fi/contracts/withdrawal-queue-erc721/
+     * Note: There is a 1k amount limit. Caller should split large withdrawals in chunks of less or equal to 1k each.)
+     */
+    function requestLidoWithdrawals(uint256[] memory amounts)
+        external
+        onlyOperatorOrOwner
+        returns (uint256[] memory requestIds)
+    {
+        requestIds = lidoWithdrawalQueue.requestWithdrawals(amounts, address(this));
+
+        // Sum the total amount of stETH being withdraw
+        uint256 totalAmountRequested = 0;
+        for (uint256 i = 0; i < amounts.length; i++) {
+            totalAmountRequested += amounts[i];
+        }
+
+        // Increase the Ether outstanding from the Lido Withdrawal Queue
+        lidoWithdrawalQueueAmount += totalAmountRequested;
+
+        emit RequestLidoWithdrawals(amounts, requestIds);
+    }
+
+    /**
+     * @notice Claim the ETH owed from the redemption requests and convert it to WETH.
+     * Before calling this method, caller should check on the request NFTs to ensure the withdrawal was processed.
+     */
+    function claimLidoWithdrawals(uint256[] memory requestIds) external {
+        uint256 etherBefore = address(this).balance;
+
+        // Claim the NFTs for ETH.
+        uint256 lastIndex = lidoWithdrawalQueue.getLastCheckpointIndex();
+        uint256[] memory hintIds = lidoWithdrawalQueue.findCheckpointHints(requestIds, 1, lastIndex);
+        lidoWithdrawalQueue.claimWithdrawals(requestIds, hintIds);
+
+        uint256 etherAfter = address(this).balance;
+
+        // Reduce the Ether outstanding from the Lido Withdrawal Queue
+        lidoWithdrawalQueueAmount -= etherAfter - etherBefore;
+
+        // Wrap all the received ETH to WETH.
+        weth.deposit{value: etherAfter}();
+
+        emit ClaimLidoWithdrawals(requestIds);
```

Comment:
🟠 We have the amount out in the `RequestLidoWithdrawals` event. It would seem symmetrical to also have it in the `ClaimLidoWithdrawals`. Plus it allows tracking the `lidoWithdrawalQueueAmount` via only events.